yt/frontends/cf_radial/data_structures.py

```diff
@@ -0,0 +1,309 @@
+                            "loading existing re-gridded file: %s", storage_filename
+                        )
+                        regrid = False
+
+                if regrid:
+                    mylog.info("Building cfradial grid")
+                    from yt.utilities.on_demand_imports import _pyart as pyart
+
+                    radar = pyart.io.read_cfradial(filename)
+
+                    grid_shape, grid_limits = self._validate_grid(
+                        radar, grid_shape, grid_limit_x, grid_limit_y, grid_limit_z
+                    )
+
+                    # grid_shape and grid_limits are in (z, y, x) order.
+                    self.grid_shape = grid_shape
+                    self.grid_limits = grid_limits
+                    mylog.info("Calling pyart.map.grid_from_radars ... ")
+                    # this is fairly slow
+                    grid = pyart.map.grid_from_radars(
+                        (radar,),
+                        grid_shape=self.grid_shape,
+                        grid_limits=self.grid_limits,
+                    )
+                    mylog.info(
+                        "Successfully built cfradial grid, writing to %s",
+                        storage_filename,
+                    )
+                    mylog.info(
+                        "Subsequent loads of %s will load the gridded file by default",
+                        filename,
+                    )
+                    grid.write(storage_filename)
+
+                filename = storage_filename
+
+        super().__init__(filename, dataset_type, units_override=units_override)
+        self.storage_filename = storage_filename
+        self.refine_by = 2  # refinement factor between a grid and its subgrid
+
+    @contextlib.contextmanager
+    def _handle(self, filename: Optional[str] = None):
+        if filename is None:
+            if hasattr(self, "filename"):
+                filename = self.filename
+            else:
+                raise RuntimeError("Dataset has no filename yet.")
+
+        with xr.open_dataset(filename) as xrds:
+            yield xrds
+
+    def _validate_grid(
+        self,
+        radar,
+        grid_shape: Optional[Tuple[int, int, int]] = None,
+        grid_limit_x: Optional[Tuple[float, float]] = None,
+        grid_limit_y: Optional[Tuple[float, float]] = None,
+        grid_limit_z: Optional[Tuple[float, float]] = None,
+    ) -> Tuple[Tuple, Tuple]:
+        # this function sets all the re-gridding parameters if they are not set
+
+        # radar is a pyart radar handle, from pyart.io.read_cfradial()
+        max_range_meters = radar.range["data"].max()  # along-ray distance
+        elevation = radar.elevation["data"]  # angle above horizontal
+        if radar.elevation["units"] == "degrees":
+            elevation = elevation * np.pi / 180.0
+        max_height = max_range_meters * np.sin(elevation.max())
+        max_distance = max_range_meters * np.cos(elevation.max())
```

Comment:
@zssherman this bit is what I'm not sure of...

The test dataset we have only has a single value for altitude:

```python
import pyart
ds_pyart = pyart.io.read_cfradial("swx_20120520_0641.nc")
print(ds_pyart.altitude)

{'units': 'meters',
 'standard_name': 'Altitude',
 'data': masked_array(data=[214.],
              mask=False,
        fill_value=1e+20)}
```
I thought maybe this was the instrument altitude, not the measurement coordinate?

And based on my reading of the coordinate info in `ds_pyart.info()`, poking around the pyart docs and skimming the CFRadial file format documentation, I understood elevation, azimuth and range to be a spherical coordinate system.. and so I ignored the azimuth and just used the extents of range and elevation to calculate the z and x/y ranges. Could **totally** be wrong about all of this though :)